src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -1035,11 +1035,18 @@ [Value] [nvarchar](max) NULL
                         }
                     }
 
-                    int counter = 0;
-                    while (counter < 10)
+                    SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString);
+                    builder.PersistSecurityInfo = true;
+                    builder.Pooling = false;
```

Comment:
I hade the most luck reproducing issues by using pooling for just 2-3 times. Disabling pooling hides some backend errors as the connection is closed anyway.